### PR TITLE
URI encode path params

### PR DESCRIPTION
**Pull request by Shahab96**
* Update test snapshot to reflect uri encoded path params
* Update method template to uri encode path params supplied by consumers
* Run npm audit fix to fix some security vulnerabilities
* Bumped version to 2.0.2

This commit fixes issue: #92

### CODE REVIEW

**Comment by Shahab96** [comment]
I'm unsure as to whether this should be a patch or a minor version bump, as now that path parameters will be uri encoded, any consumers who are manually encoding their path parameters will be broken by this change.

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "@types/mustache": "^0.8.32",
     "@types/node": "^10.12.18",
     "final-fs": "^1.6.0",
-    "grunt": "^1.0.3",
+    "grunt": "^1.0.4",
```
Comment (mtennoe) [inline package.json]: Lets keep the versions of the dependencies as is

File: templates/method.mustache [hunk 2]
```diff
@@ -106,7 +106,7 @@
     {{/isQueryParameter}}
 
     {{#isPathParameter}}
-        path = path.replace('{{=<% %>=}}{<%&name%>}<%={{ }}=%>', `${parameters['{{&camelCaseName}}']}`);
+        path = path.replace('{{=<% %>=}}{<%&name%>}<%={{ }}=%>', `${encodeURIComponent(parameters['{{&camelCaseName}}'].toString())}`);
```
Comment (mtennoe) [inline templates/method.mustache]: Is the tostring() necessary?

**Comment by mtennoe** [comment]
Re. backwards compatibility - That is a very good question. For backwards compatiblity breaking changes we want to bump the major version, but this is more of a small bugfix, so a major version bump seems drastic. I am leaning towards a major bump to "do it right" though. @Markionium - Thoughts?

File: templates/method.mustache (hunk 2 above)
Comment (Shahab96) [reply templates/method.mustache]: The toString is required in case the input is a number or such, as encodeURIComponent requires string input.

Comment (mtennoe) [reply templates/method.mustache]: Makes sense!

**Comment by mtennoe** [comment]
@Shahab96 - This looks good! Before I merge, let's be safe and bump it to 3.0.0. If you want to you can check it in as the existing version (2.0.1), and I can bump it to 3 when I publish it

**Comment by Shahab96** [comment]
> @Shahab96 - This looks good! Before I merge, let's be safe and bump it to 3.0.0. If you want to you can check it in as the existing version (2.0.1), and I can bump it to 3 when I publish it

I can just bump the version to 3.0.0. No problem :)

**Comment by mtennoe** [comment]
@Shahab96 - Merged and published as 3.0.0. I also did a quick npm audit fix to bump the vulnerable packages, published as 3.0.1

**Pull request merged**